### PR TITLE
Build based on HiGHS_jll 0.2

### LINKED ISSUE #13: Update to HiGHS_jll 0.2
Now building on Windows: https://github.com/JuliaBinaryWrappers/HiGHS_jll.jl/releases/tag/HiGHS-v0.2.0%2B0

### PR DESCRIPTION
**Pull request by odow**
This PR is horribly breaking for anyone using the C API, so I hope there isn't anyone. It also:
* Adds support for HiGHS_jll 0.2.
* Simplifies the `gen` script, including compiling it directly from the `HiGHS_jll` install.
* Removes `CWrapper`.
* Removes `ManagedHiGHS` in favor of managing the `Optimizer` object directly.
* Uses `cconvert` and `unsafe_convert` to allow passing `Optimizer` to C functions directly, instead of passing the pointer.
* Exports all `Highs_xxx` function from the C API.
* Simplifies option handling.
* Fills out the README content.

<s>This is branched from #14. I can rebase it after it is merged. I guess for now, the tests will fail.
 
I have a few comments, based on learnings from wrapping the other solvers.

- The less helper/wrapper/intermediate functions/types/files/modules we have the better. There should be the C API directly from `HiGHS_jll`, and there should be the MOI wrapper. 
  - Intermediate functions such as https://github.com/jump-dev/HiGHS.jl/blob/b42183d1f44da67c553a1ff63162832a1925483e/src/wrapper/api_helpers.jl#L13-L15 just add complexity over time. The MOI wrapper can abstract the difficulty of handling different option types in RawParameter. We don't need a way of doing this for the C API.
  - If there is no intermediate API, then code can be exactly the same as the C API. This _greatly_ aids the reproducibility of upstream bugs.
  - The namespace justification doesn't apply, because every C function already has a namespace by starting with `Highs_`. And the MOI functions should all extend `MOI.`, or begin with an underscore.
</s>

Closes #13

### CODE REVIEW

**Comment by odow** [comment]
There are a lot of improvements that can be made to the MOI wrapper, but this PR at least gets it to a good point for future changes.

The only real API issue that could be helpful to change upstream is a way of querying the type of an option given the string name. For now, I've just used a dictionary: 
https://github.com/jump-dev/HiGHS.jl/pull/15/files#diff-64cebae681efceded214317cd18b678aaefed5658af7a123f9b01433313a2844R163-R172
but it would be helpful to have an equivalent of Gurobi's (undocumented) `GRBgetparamtype`, which returns an `int` corresponding to the parameter type (-1 = unsupported, 1 = int, 2 = double, 3 = string). Thoughts @galabovaa?

File: gen/gen.jl [hunk 1]
```diff
@@ -1,52 +1,24 @@
...
+const LIBHIGHS_HEADERS = [
+    joinpath(HiGHS_jll.artifact_dir, "include", "interfaces", "highs_c_api.h"),
+]
 
-const header_file = joinpath(HIGHS_DIR, "src/interfaces/highs_c_api.h")
-
-const LIB_HEADERS = [header_file]
-
-const ctx = Clang.DefaultContext()
+const GEN_DIR = joinpath(dirname(@__DIR__), "src", "wrapper")
```
Comment (matbesancon) [inline gen/gen.jl:8]:
```suggestion
const GEN_DIR = joinpath(@__DIR__, "..", "src", "wrapper")
```
dirname of a dir just extracts the directory above, I find it more explicit with `..`

**Comment by matbesancon** [comment]
Looks great. One remark on the generation phase, it should ideally be possible to do it not from the HIGHS_jll, but also from a custom HiGHS install since this will likely be done by the developers after modifying HiGHS itself, hence the HIGHS_DIR. If not provided, we can fall back to the HiGHS_jll one

**Comment by galabovaa** [comment]
We can add such an option to HiGHS as GRBgetparamtype for the types of options. 

The simplex code is going through a refactoring and once that is merged I can do that. Probably within the next month.

**Comment by odow** [comment]
I used `HiGHS_jll` for the code generation phase because HiGHS.jl doesn't support local installs, and so it doesn't make sense to generate the code from a potentially different version.

The steps for updating the wrapper would be: release HiGHS, modify Yggdrasil and release a new jll, modify HiGHS.jl.

We can change back later in another PR if this isn't working well.

**Comment by matbesancon** [comment]
> because HiGHS.jl doesn't support local installs

you're right, I had forgotten that I completely removed the install from custom HiGHS dir, this is something I'll re-add in another PR, I think this can be useful. For now this is totally fine as-is. Furthermore, the generator is likely less required to need regeneration

**Comment by odow** [comment]
I propose we merge this so we can start some follow-up PRs.

**Review by matbesancon — APPROVED** [review]

**Pull request merged**